Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Driver.hs

```diff
@@ -1533,25 +1533,28 @@ sendSmsToDriver merchantShortId opCity driverId volunteerId _req@SendSmsReq {..}
       sender = smsCfg.sender
   withLogTag ("personId_" <> personId.getId) $ do
     case channel of
-      SMS -> whenJust messageKey $ \mkey -> do
+      SMS -> do
+        mkey <- fromMaybeM (InvalidRequest "Message Key field is required for channel : SMS") messageKey --whenJust messageKey $ \mkey -> do
         message <- MessageBuilder.buildGenericMessage merchantOpCityId mkey MessageBuilder.BuildGenericMessageReq {}
         Sms.sendSMS driver.merchantId merchantOpCityId (Sms.SendSMSReq message phoneNumber sender)
           >>= Sms.checkSmsResult
-      WHATSAPP -> whenJust messageKey $ \mkey -> do
+      WHATSAPP -> do
+        mkey <- fromMaybeM (InvalidRequest "Message Key field is required for channel : WHATSAPP") messageKey -- whenJust messageKey $ \mkey -> do
         merchantMessage <-
           QMM.findByMerchantOpCityIdAndMessageKey merchantOpCityId mkey
             >>= fromMaybeM (MerchantMessageNotFound merchantOpCityId.getId (show mkey))
         let jsonData = merchantMessage.jsonData
         result <- Whatsapp.whatsAppSendMessageWithTemplateIdAPI driver.merchantId merchantOpCityId (Whatsapp.SendWhatsAppMessageWithTemplateIdApIReq phoneNumber merchantMessage.templateId jsonData.var1 jsonData.var2 jsonData.var3 (Just merchantMessage.containsUrlButton))
         when (result._response.status /= "success") $ throwError (InternalError "Unable to send Whatsapp message via dashboard")
-      OVERLAY -> whenJust overlayKey $ \oKey -> do
+      OVERLAY -> do
+        oKey <- fromMaybeM (InvalidRequest "Overlay Key field is required for channel : OVERLAY") overlayKey --whenJust overlayKey $ \oKey -> do
```

Comment:
commented code

Reply:
I have commented the previous code.. for the case if we want to revert to the previous change while refactoring the error codes for solving 4xx

Reply:
Will remove this commented code, once refactoring is done